internal/xds/bootstrap/tlscreds/bundle.go

```diff
@@ -64,6 +65,9 @@ func NewBundle(jd json.RawMessage) (credentials.Bundle, func(), error) {
 		}
 	} // Else the config field is absent. Treat it as an empty config.
 
+	if !envconfig.XDSSpiffeEnabled {
```

Comment:
This functionality was added as part of A65, right? https://github.com/grpc/proposal/blob/master/A65-xds-mtls-creds-in-bootstrap.md

And that gRFC says that we will not guard this with an environment variable. Am I missing something?

Even if we think we need to guard here with the same env variable, we probably need to do the same thing that I mentioned in the other comment. i.e., not even read the field if the env var is set to false.